### PR TITLE
feat(zoe): multiples, for making divisible offers

### PR DESCRIPTION
**Pull request by erights**
OMG Multiples! But tested only that we didn't break anything, because everything currently uses only `multiples: 1n`.

refs: #403, #5540

@Chris-Hibbert @dtribble @mhofman @turadg , once ready for review, I will ask you to review. Until then, please ignore. (@turadg , I cannot unlist you because of codeowners. @Chris-Hibbert I cannot and should not unlist you since you already commented.)

(Attempt https://github.com/Agoric/agoric-sdk/pull/11386 to bring this up to date failed)

### CODE REVIEW

**Comment by Chris-Hibbert** [comment]
This is nice!   I think it will work for the most obvious cases.

I don't know how to think about the corner cases without an outline of the expected usage pattern. With that, someone other than you (@erights) could conceivably finish this.

**Comment by erights** [comment]
> This is nice! I think it will work for the most obvious cases.
> 
> I don't know how to think about the corner cases without an outline of the expected usage pattern. With that, someone other than you (@erights) could conceivably finish this.

@turadg @dtribble would that be one of you?

**Comment by turadg** [comment]
> @turadg @dtribble would that be one of you?

SGTM! This is quite interesting. I'll have to sequence it with MN-1 priorities. Is this aiming for that release?
 
> I don't know how to think about the corner cases without an outline of the expected usage pattern.

Same here, I would need at least some description of tests of multiples > 1.

File: packages/zoe/src/cleanProposal.js [hunk 1]
```diff
@@ -162,10 +164,41 @@ export const cleanProposal = (proposal, getAssetKindByBrand) => {
...
+    return amount;
+  }
+  const { brand, value } = amount;
+  assert(value >= 1n);
+  assert.typeof(
+    value,
+    'bigint',
+    X`multiples > 1 not yet implemented for non-fungibles: ${multiples} * ${amount}`,
+  );
+  return harden({ brand, value: value * multiples });
```
Comment (Chris-Hibbert) [inline packages/zoe/src/cleanProposal.js]: don't we use natSafeMath even for addition and multiplication?

File: packages/zoe/src/contractFacet/offerSafety.js [hunk 2]
```diff
@@ -2,25 +2,41 @@
...
     }
     const allocationAmount = allocation[keyword];
-    return AmountMath.isGTE(allocationAmount, requiredAmount);
-  };
-  return Object.entries(giveOrWant).every(isGTEByKeyword);
+    if (!AmountMath.isGTE(allocationAmount, requiredAmount)) {
+      return 0n;
+    }
+    if (typeof requiredAmount.value !== 'bigint') {
+      multiples = 1n;
```
Comment (Chris-Hibbert) [inline packages/zoe/src/contractFacet/offerSafety.js]: too much inconsistency between immediate returns and setting `multiples`.  Here I'd `return 1n`.

Comment (erights) [reply packages/zoe/src/contractFacet/offerSafety.js]: The `multiples` here are being set within the loop, whereas the `multiples` variable is being set outside the loop. If on one iteration this condition applies, it'll set `multiples` to `1n`. If on a later iteration the condition above it applies, then the function as a whole will return `0n`. If we returned here, the function as a whole would return `1n` instead, which would be wrong.

File: packages/zoe/src/cleanProposal.js (hunk 1 above)
Comment (erights) [reply packages/zoe/src/cleanProposal.js]: Done. Also for the divide in offerSafety.js where it was even more valuable because it explicitly states it is doing a floorDivide, which is important for any reader to know.
